packages/eslint-plugin/tests/rules/consistent-type-imports.test.ts

```diff
@@ -0,0 +1,250 @@
+import rule from '../../src/rules/consistent-type-imports';
+import { RuleTester } from '../RuleTester';
+
+const ruleTester = new RuleTester({
+  parser: '@typescript-eslint/parser',
+  parserOptions: {
+    ecmaVersion: 2020,
+    sourceType: 'module',
+  },
+});
+
+ruleTester.run('consistent-type-imports', rule, {
```

Comment:
<details>
<summary>some extra test cases that you'll want to test</summary>

```ts
import Type from 'foo';

type T = typeof Type;
type T = typeof Type.foo;
```

```ts
import type Type from 'foo';

type T = typeof Type;
type T = typeof Type.foo;
```

```ts
import { Type } from 'foo';

type T = typeof Type;
type T = typeof Type.foo;
```

```ts
import type { Type } from 'foo';

type T = typeof Type;
type T = typeof Type.foo;
```

```ts
import * as Type from 'foo';

type T = typeof Type;
type T = typeof Type.foo;
```

```ts
import type * as Type from 'foo';

type T = typeof Type;
type T = typeof Type.foo;
```

</details>

Plus the examples I listed in the comments in the suggestion [here](#TODO)